src/notevennexus/tree.py

```diff
@@ -0,0 +1,38 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Any
+
+
+@dataclass
+class Dataset:
+    """Info about an HDF5 dataset"""
+
+    name: str
+    shape: tuple[int, ...]
+    dtype: str
+    parent: Group
+    attrs: dict[str, Any] = field(default_factory=dict)
+    value: Any | None = None
+
+
+@dataclass
+class Group:
+    """Info about an HDF5 group"""
+
+    name: str
+    parent: Group | None = None
+    attrs: dict[str, Any] = field(default_factory=dict)
+    children: dict[str, Dataset | Group] = field(default_factory=dict)
+
+
+def unroll_tree(tree: Group) -> dict[str, Dataset | Group]:
+    """Unroll tree into a flat dictionary"""
+    result = {}
+    for child in tree.children.values():
+        result[child.name] = child
```

Comment:
Are the child names always unique? Could there be a case in a badly mad nexus file where the name would not be unique and you would be missing some entries in the results?

Reply:
In HDF5? No, how? Maybe one could write some JSON that does this, but then we should fail in the loader, not here.